### PR TITLE
standalone rpc_types

### PR DESCRIPTION
**Pull request by supernovahs**
Ref #3964

### CODE REVIEW

**Comment by supernovahs** [comment]
Will now  figure out how to delete the duplicate code from https://github.com/paradigmxyz/reth/blob/c423514321c2d2e67a156681d5602cb1707852e3/crates/rpc/rpc-types/src/eth/transaction/mod.rs

File: Cargo.toml [hunk 1]
```diff
@@ -45,7 +45,7 @@ members = [
     "crates/transaction-pool",
     "crates/trie",
     "testing/ef-tests",
-
+    "crates/rpc/rpc-types-util",
```
Comment (mattsse) [inline Cargo.toml:48]: I would prefer something other than util

File: crates/rpc/rpc-types-util/src/transaction/mod.rs [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+};
+
+use reth_rpc_types::Signature;
+
+use serde::{Deserialize, Serialize};
+
+/// Transaction object used in RPC
+#[derive(Debug, Default, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Transaction {
```
Comment (mattsse) [inline crates/rpc/rpc-types-util/src/transaction/mod.rs:15]: why was this type moved?
this is an rpc type and should remain in rpc-types

**Review by mattsse — CHANGES_REQUESTED** [review]
perhaps there was a misunderstanding,

all type defs should remain in rpc-types crate, and we only want to extract all the functions

right @DaniPopes ?

File: crates/rpc/rpc-types-util/src/transaction/mod.rs (hunk 2 above)
Comment (supernovahs) [reply crates/rpc/rpc-types-util/src/transaction/mod.rs:15]: Right

File: Cargo.toml (hunk 1 above)
Comment (supernovahs) [reply Cargo.toml:48]: ok

File: crates/rpc/rpc-types-util/src/transaction/mod.rs (hunk 2 above)
Comment (supernovahs) [reply crates/rpc/rpc-types-util/src/transaction/mod.rs:15]: Also, I copied the type , not moved .

File: Cargo.toml (hunk 1 above)
Comment (mattsse) [reply Cargo.toml:48]: I think `compat` or `convert` are better alternatives

**Comment by supernovahs** [comment]
Don't know why I am getting circular  dependency errors
```
error: cyclic package dependency: package `reth-rpc-types v0.1.0-alpha.4 (/Users/supernovahs/Desktop/reth/crates/rpc/rpc-types)` depends on itself. Cycle:
package `reth-rpc-types v0.1.0-alpha.4 (/Users/supernovahs/Desktop/reth/crates/rpc/rpc-types)`
    ... which satisfies path dependency `reth-rpc-types` of package `reth-rpc-types-compat v0.1.0-alpha.4 (/Users/supernovahs/Desktop/reth/crates/rpc/rpc-types-compat)`
    ... which satisfies path dependency `reth-rpc-types-compat` of package `reth-rpc-types v0.1.0-alpha.4 (/Users/supernovahs/Desktop/reth/crates/rpc/rpc-types)`
    ... which satisfies path dependency `reth-rpc-types` of package `reth-network-api v0.1.0-alpha.4 (/Users/supernovahs/Desktop/reth/crates/net/network-api)`
    ... which satisfies path dependency `reth-network-api` of package `reth-interfaces v0.1.0-alpha.4 (/Users/supernovahs/Desktop/reth/crates/interfaces)`
    ... which satisfies path dependency `reth-interfaces` of package `reth-consensus-common v0.1.0-alpha.4 (/Users/supernovahs/Desktop/reth/crates/consensus/common)`
    ... which satisfies path dependency `reth-consensus-common` of package `reth-beacon-consensus v0.1.0-alpha.4 (/Users/supernovahs/Desktop/reth/crates/consensus/beacon)`
    ... which satisfies path dependency `reth-beacon-consensus` of package `reth-auto-seal-consensus v0.1.0-alpha.4 (/Users/supernovahs/Desktop/reth/crates/consensus/auto-seal)`
    ... which satisfies path dependency `reth-auto-seal-consensus` of package `reth v0.1.0-alpha.4 (/Users/supernovahs/Desktop/reth/bin/reth)`
    ... which satisfies path dependency `reth` of package `additional-rpc-namespace-in-cli v0.0.0 (/Users/supernovahs/Desktop/reth/examples/additional-rpc-namespace-in-cli)`
```

**Comment by supernovahs** [comment]
Got it, I would have to separate every function , such that rpc_types does not import compat in any case

**Comment by supernovahs** [comment]
I have made the changes. But +nightly fmt could not be done, due to some issues with my mac .
Ty

**Review by mattsse — APPROVED** [review]
amazing, this looks great, ty!

I'll take care of the fmt issue

**Pull request merged**